wire-ios-request-strategy/Sources/Payloads/Processing/Helpers/MLSEventProcessor.swift

```diff
@@ -78,7 +78,7 @@ public class MLSEventProcessor: MLSEventProcessing {
             return logWarn(aborting: .conversationUpdate, withReason: .conversationNotMLSCapable)
         }
 
-        guard let mlsGroupID = mlsGroupID ?? MLSGroupID(from: groupID) else {
+        guard let mlsGroupID = mlsGroupID ?? groupID.map({ MLSGroupID(base64Encoded: $0) }) ?? .none else {
```

Comment:
Somehow the func here seems to have multiple responsibilities. I would rather split it in smaller funcs that run the same logic in a private func.
E.g. I ask myself when reading it: why should I pass an optional `groupID` if it is ignored in case I can find the `mlsGroupID`?

Reply:
I agree, it is also not very clear to me.
I renamed the argument `fallbackGroupID`.